stratum/hal/lib/barefoot/bf_sde_wrapper.cc

```diff
@@ -562,15 +567,68 @@ ::util::Status TableData::SetActionMemberId(uint64 action_member_id) {
 }
 
 ::util::Status TableData::GetActionMemberId(uint64* action_member_id) const {
-  return GetField(*table_data_, "$ACTION_MEMBER_ID", action_member_id);
+  const bfrt::BfRtTable* table;
+  RETURN_IF_BFRT_ERROR(table_data_->getParent(&table));
+  bf_rt_id_t field_id;
+  bfrt::DataType data_type;
+  if (table->actionIdApplicable()) {
+    bf_rt_id_t action_id;
+    RETURN_IF_BFRT_ERROR(table_data_->actionIdGet(&action_id));
+    RETURN_IF_BFRT_ERROR_WITHOUT_LOGGING(
+        table->dataFieldIdGet("$ACTION_MEMBER_ID", action_id, &field_id));
+    RETURN_IF_BFRT_ERROR(
+        table->dataFieldDataTypeGet(field_id, action_id, &data_type));
+  } else {
+    RETURN_IF_BFRT_ERROR(table->dataFieldIdGet("$ACTION_MEMBER_ID", &field_id));
```

Comment:
Should be `RETURN_IF_BFRT_ERROR_WITHOUT_LOGGING`?

Reply:
This is fine because if the table as no action IDs, it's usually an indirect one.